src/citrine/builders/design_spaces.py

```diff
@@ -0,0 +1,96 @@
+import pandas as pd
+from itertools import product
+
+def EnumeratedProductDesignSpaceData(dof_grids:dict):
+    ''' Given a dict of lists or tuples of values for each degree of freedom,
+    create the list of candidates representing the outer product of all values
+    for each degree of freedom
+    
+    Parameters
+    ----------
+    dof_grids: dict(str:list)
+        {'degree of freedom':[list, of, values]} for each DoF
+    '''
+    design_space_tuples = list(product(*dof_grids.values()))
+    design_space_cols = list(dof_grids.keys())
+    df_ds = pd.DataFrame(data=design_space_tuples, columns=design_space_cols)
+    candidates = df_ds.to_dict('records')
+    return candidates
+
+
+def SimpleMixtureProductDesignSpaceData(formulation_grids:dict,
```

Comment:
```suggestion
def enumerate_simple_mixture_from_ingredients(formulation_grids:dict,
```
Again, `snake_case` and verb-y.